graphql-protocol/src/main/java/com/linecorp/armeria/server/graphql/protocol/AbstractGraphqlService.java

```diff
@@ -91,6 +99,52 @@ protected HttpResponse doPost(ServiceRequestContext ctx, HttpRequest request) th
             return unsupportedMediaType();
         }
 
+        // See https://github.com/jaydenseric/graphql-multipart-request-spec/blob/master/readme.md
+        if (contentType.is(MediaType.MULTIPART_FORM_DATA)) {
+            return HttpResponse.from(FileAggregatedMultipart.aggregateMultipart(ctx, request)
+                                                            .thenApply(multipart -> {
+                try {
+                    final String operationsParam = toStringFromMultipartParam("operations", multipart.params());
+                    final String mapParam = toStringFromMultipartParam("map", multipart.params());
+
+                    final Map<String, Object> operation = parseJsonString(operationsParam, JSON_MAP);
+                    final Map<String, List<String>> map = parseJsonString(mapParam, MAP_PARAM);
+                    final String query = toStringFromJson("query", operation.get("query"));
+                    if (Strings.isNullOrEmpty(query)) {
+                        return HttpResponse.of(HttpStatus.BAD_REQUEST, MediaType.PLAIN_TEXT,
+                                               "Missing query");
+                    }
+                    final String operationName =
+                            toStringFromJson("operationName", operation.get("operationName"));
+                    final Map<String, Object> variables = toMapFromJson(operation.get("variables"));
```

Comment:
According to the GraphQL multipart specification, `variables` and `map` seem not to be null or empty. 
Should we return a bad request response for that?

Reply:
I thought this should be optional for the user to use. Wouldn't there be a situation where it should succeed even if the file is empty? 🤔

Reply:
I see. Let's keep it as is.

Reply:
I see. Let's keep it as is.